[
  {
    "set": "bc",
    "id": 1,
    "messages": [
      {
        "msg": "usage",
        "id": 1,
        "description": "Usage message for the program. {0} is the program name.",
        "C": "Usage: {0} [-l] [file...]"
      },
      {
        "msg": "unrecognised_option",
        "id": 2,
        "description": "Option -{0} is unrecognised.",
        "C": "Option '-{0}' is not valid."
      },
      {
        "msg": "file_error",
        "id": 3,
        "description": "Error whilst parsing file {0}, aborting.",
        "C": "Abort: File system error whilst parsing '{0}'."
      },
      {
        "msg": "unexpected_token",
        "id": 4,
        "description": "Unexpected token {0}",
        "C": "Unexpected token: '{0}'"
      },
      {
        "msg": "error_label",
        "id": 5,
        "description": "How to label an error",
        "C": "Error"
      },
      {
        "msg": "unexpected_eof_string",
        "id": 6,
        "description": "Unexpected End-of-file whilst parsing a string",
        "C": "Unexpected end of file encountered, whilst parsing a string."
      },
      {
        "msg": "unexpected_eof_comment",
        "id": 7,
        "description": "Unexpected End-of-file whilst parsing a comment",
        "C": "Unexpected end of file encountered, whilst parsing a comment."
      },
      {
        "msg": "digit_out_of_range",
        "id": 8,
        "description": "The digit {0} is not in range for the base {1}",
        "C": "Numeric digit {0} is not valid in base {1}."
      },
      {
        "msg": "assign_minus_undefined",
        "id": 9,
        "description": "The token =- results in underfined behaviour, report it as an error.",
        "C": "Token '=-' is not recognised."
      },
      {
        "msg": "expected_statement",
        "id": 10,
        "description": "We expected a statement but instead got the token {0}",
        "C": "Expected <statement>, found: {0}"
      },
      {
        "msg": "expected_rbrace",
        "id": 11,
        "description": "We expected a } but instead got the token {0}",
        "C": "Expected '}}', found: {0}"
      },
      {
        "msg": "missing_newline_at_end_of_semicolon_list",
        "id": 12,
        "description": "Expected a new line at the end of a semicolon list",
        "C": "Expected new-line, found: {0}"
      },
      {
        "msg": "expected_expression",
        "id": 13,
        "description": "Expected an expression, found: {0}",
        "C": "Expected <expression>, found: {0}"
      },
      {
        "msg": "expected_assign_expression",
        "id": 14,
        "description": "Expected an assignment expression, found: {0}",
        "C": "Expected <assignment expression>, found: {0}"
      },
      {
        "msg": "expected_named_expression",
        "id": 15,
        "description": "Expected a named expression, found: {0}",
        "C": "Expected variable or array access, found: {0}"
      },
      {
        "msg": "expected_rsquare",
        "id": 16,
        "description": "Expected ']', found: {0}",
        "C": "Expected ']', found: {0}"
      },
      {
        "msg": "expected_lparens",
        "id": 17,
        "description": "Expected '(', found: {0}",
        "C": "Expected '(', found: {0}"
      },
      {
        "msg": "expected_rparens",
        "id": 18,
        "description": "Expected a ')', found: {0}",
        "C": "Expected ')', found: {0}"
      },
      {
        "msg": "expected_primary_expression",
        "id": 19,
        "description": "Expected a primary expression, found: {0}",
        "C": "Expected primary expression, found: {0}"
      },
      {
        "msg": "break_outside_of_loop",
        "id": 20,
        "description": "Encountered a break when not in a loop",
        "C": "Encountered break whilst not in a for/while loop."
      },
      {
        "msg": "expected_semicolon",
        "id": 21,
        "description": "Expected an ';', found: {0}",
        "C": "Expected ';', found: {0}"
      },
      {
        "msg": "return_outside_function",
        "id": 22,
        "description": "Encountered a return statement outside of a function.",
        "C": "Encountered a return statement outside of a function."
      },
      {
        "msg": "expected_letter",
        "id": 23,
        "description": "Expected a letter, found: {0}",
        "C": "Expected <letter>, found: {0}"
      },
      {
        "msg": "expected_lbrace",
        "id": 24,
        "description": "Expected a '{', found: {0}",
        "C": "Expected '{', found: {0}"
      },
      {
        "msg": "expected_newline",
        "id": 25,
        "description": "Expected a newline, found: {0}",
        "C": "Expected a newline, found: {0}"
      },
      {
        "msg": "unable_to_nest_function_definitions",
        "id": 26,
        "description": "Not allowed to nest function definitions",
        "C": "Can't define one function inside another."
      },
      {
        "msg": "expected_semicolon_or_newline",
        "id": 27,
        "description": "Expected a semicolon or newline, found: {0}",
        "C": "Expected a semicolon or newline, found: {0}"
      },
      {
        "msg": "duplicate_variables_in_function_definition",
        "id": 28,
        "description": "Variable {0} appears more than once in the definition of a function.",
        "C": "Repeated variable names in function definition: {0}"
      },
      {
        "msg": "duplicate_arrays_in_function_definition",
        "id": 29,
        "description": "Array {0} appears more than once in the definition of a function.",
        "C": "Repeated array names in function definition: {0}[]"
      },
      {
        "msg": "expected_define_list",
        "id": 30,
        "description": "Expected define list not: {0}",
        "C": "Expected a list of variable and array names, found: {0}"
      },
      {
        "msg": "expected_define_element",
        "id": 31,
        "description": "Expected define element not: {0}",
        "C": "Expected a variable or array name, found: {0}"
      },
      {
        "msg": "base_out_of_range",
        "id": 32,
        "description": "Special variable {0} (ibase/obase - names are not translated) is out of range, should be [2, 16] we got {1}",
        "C": "{0} set to an out of range value {1}.  Should be in range 2 <= {0} <= 16."
      },
      {
        "msg": "number_to_unsigned_failed_negative",
        "id": 33,
        "description": "Failed to convert number {0} to unsigned integer as it was negative.",
        "C": "Cannot convert {0} to unsigned integer, as it is negative."
      },
      {
        "msg": "number_to_unsigned_failed_fractional",
        "id": 34,
        "description": "Failed to convert number {0} to unsigned integer as it was fractional.",
        "C": "Cannot convert {0} to unsigned integer, as it has a fraction component."
      },
      {
        "msg": "number_to_unsigned_failed_too_large",
        "id": 35,
        "description": "Failed to convert number {0} to unsigned integer as it is too big.",
        "C": "Cannot convert {0} to unsigned integer, as it is larger than the underlying type."
      },
      {
        "msg": "no_end_to_function_definition",
        "id": 36,
        "description": "Error:{0}:{1}:{2}: Function defined at location has no end of definition..",
        "C": "Error:{0}:{1}:{2}: No end to function definition."
      }
    ]
  }
]